src/core/compiler.jl

```diff
@@ -228,14 +237,38 @@ function build_model_info(input_expr)
     warn_empty(modeldef[:body])
     # Construct model_info dictionary
 
-    arg_syms = [(arg isa Symbol) ? arg : arg.args[1] for arg in modeldef[:args]]
+    arg_syms = map(modeldef[:args]) do arg
```

Comment:
Can you give me a summary here of what the following changes in doing?

Reply:
I am extracting all the argument symbols. When the argument is `demo(x)` (branch 1), `demo(::Type{T}) where {T}` (branch 2) or `demo(x = 1)` (branch 3).

Reply:
Got it. Maybe make this as a comment.

Reply:
Sure thing!